adal/src/main/java/com/microsoft/aad/adal/AcquireTokenSilentHandler.java

```diff
@@ -129,6 +130,24 @@ AuthenticationResult acquireTokenWithRefreshToken(final String refreshToken)
                 result.setRefreshToken(refreshToken);
             }
         } catch (final IOException | AuthenticationException exc) {
+            //if get 503,504,500 && outageMode is on, return the stale token
+            if (exc instanceof AuthenticationException
```

Comment:
Then shouldn't we think if we try to minimize code change, are we introducing good design or not?
Here is something I noted down before:
From Effective C++, by Scott Meyers :
"Anytime you find yourself writing code of the form "if the object is of type T1, then do something, but if it's of type T2, then do something else," slap yourself.